subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -80,16 +74,14 @@ static void state_status_u8(struct bt_mesh_model *model,
 	       ctx->net_idx, ctx->app_idx, ctx->addr, buf->len,
 	       bt_hex(buf->data, buf->len));
 
-	if (!cli_response_check(expect_status, ctx->addr)) {
-		BT_WARN("Unexpected Status (0x%08x != 0x%08x)",
-			cli->op_pending, expect_status);
+	if (!bt_mesh_model_ack_match(&cli->ack_ctx, expect_status, ctx)) {
 		return;
 	}
 
-	status = cli->op_param;
```

Comment:
Is there a reason you didn't want to follow the suggestion of having an API that returns the parameters conditionally, and instead kept the explicit `check()` function. I was thinking of something like:
```
status = bt_mesh_model_ack_param_get(&cli->ack_ctx, expect_status, ctx);
if (!status) {
        return;
}

...
```

The only downside I can think of is if you want to wait for something that doesn't have any associated parameters, but I think in every case the reason we wait is so we at least get the status (and in some cases also additional parameters).

FWIW, I'm not completely sold on the naming or passing msg_ctg instead of the address, but I'll need to let this brew in my head for awhile :)

Reply:
The reason I kept `check()` function is because if I want to set `user_data` to NULL, `bt_mesh_model_ack_param_get()` won't work anymore as it will return NULL no matter of the message.

Reply:
Yeah, that's what I meant with "doesn't have any associated parameters" in my earlier comment. Indeed, if we want to keep this future-proof, if someone just wants to use it as a synchronisation point without caring for return parameter values, we need this kind of API.

Reply:
Won't it be enough to call `bt_mesh_model_ack_rx` then? As a compromise, I can suggest the following API:
```
bool bt_mesh_cli_param_get(struct bt_mesh_cli *cli, uint32_t opcode, uint16_t addr, void **user_data);
```

Reply:
> Won't it be enough to call bt_mesh_model_ack_rx then?

Not if you also want to check that the pending operation is for a specific opcode and address first.

> As a compromise, I can suggest the following API:

I'm really fine either way, i.e. `match()` works for me with the argument provided, so I'd keep it. That said, if you want to save some code in the existing handlers then a `param_get()` (that returns the param pointer or NULL if no match) could be created by reusing `match()` internally in its implementation.